utils/src/fetch_portals/callbacks.py

```diff
@@ -0,0 +1,196 @@
+for the 5.570 brazilian municipalities.
+"""
+
+import json
+import logging
+import os
+from dataclasses import is_dataclass
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Any, Iterable, Literal, Optional, Union
+
+import pandas as pd
+
+from .models import ExistingBehavior, PathLike, PortalCapture
+
+
+def _autogen_version_notes(
+    dest_file: str, operation: Literal["create", "append", "update"]
+) -> str:
+    """Generate a default version note message.
+
+    Parameters:
+        dest_file: Destination file being written (or appended to).
+        operation: Whether the file is being ``create``'d, ``append`'ed to or
+            completely ``update``'d (replaced).
+    """
+
+    logging.warning(
+        "Version notes not provided; a default message will be generated."
+    )
+
+    if operation == "create":
+        version_notes = "Create " + dest_file
+    elif operation == "append":
+        version_notes = "Add records to " + dest_file
+    elif operation == "update":
+        version_notes = "Update " + dest_file
+
+    return version_notes
+
+
+def to_kaggle(
+    data: Union[Iterable[Union[dict, PortalCapture]], pd.DataFrame],
+    dataset: str,
+    dest_file: str,
+    existing_behavior: ExistingBehavior = "replace",
+    version_notes: Optional[str] = None,
+    local_dir: Optional[PathLike] = None,
+    delete_old_versions: bool = False,
+) -> "DatasetNewVersionResponse":  # type: ignore  # noqa: F821
+    """Write data to a destination dataset file in Kaggle.
+
+    Parameters:
+        data: Data to be uploaded to the dataset. Can be a pandas `DataFrame`_
+            instance, or an iterable of dataclass objects or dictionaries.
+        dataset: Kaggle dataset id, in the format ``<owner username>/<dataset
+            name>``.
+        dest_file: How to name the destination file in the dataset context.
+        existing_behavior: What to do if the file already exists in the
+            dataset.
+        version_notes: A message describing what changes will be made to the
+            dataset (optional; a default message will be generated if none was
+            given).
+        local_dir: A local directory where the dataset files will persist
+            (optional; defaults to None).
+        delete_old_versions: Whether to delete previous versions of the
+            dataset that exist in Kaggle.
+
+    Returns
+        A `DatasetNewVersionResponse`_ instance with the new dataset version.
+
+    .. _DataFrame: https://pandas.pydata.org/pandas-docs/stable/reference/
+            frame.html
+    .. _DatasetNewVersionResponse: https://github.com/Kaggle/kaggle-api/blob/
+        89eb72dd811492c500839f65332f669cd839d2bc/kaggle/models/
+        kaggle_models_extended.py#L150
+    """
+
+    from kaggle.api.kaggle_api_extended import KaggleApi  # type: ignore
+    from kaggle.models.kaggle_models_extended import Metadata  # type: ignore

```

Comment:
`# type: ignore` tells linters to ignore these lines, but tells nothing to fellow developers about _why_ you are importing this here (and not in the top of the file as usual). What about a friendly comment here to help fellow coders?